### PR TITLE
Add OTLP integration test for hyper and request client for logs and traces

### PR DESCRIPTION
**Pull request by lalitb**
## Changes

reqwest-blocking is failing, could be the issue with the exporter. will fix that separately.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Comment by lalitb** [comment]
Moving to draft to fix build issues

**Comment by lalitb** [comment]
> Moving to draft to fix build issues

ok fixed the build now.

File: opentelemetry-otlp/tests/integration_test/tests/logs.rs [hunk 1]
```diff
@@ -12,7 +12,18 @@ use std::fs::File;
...
-    let exporter = LogExporter::builder().with_tonic().build()?;
+    let exporter_builder = LogExporter::builder();
+    #[cfg(feature = "tonic-client")]
+    let exporter_builder = exporter_builder.with_tonic();
+    #[cfg(not(feature = "tonic-client"))]
+    #[cfg(any(
+        feature = "hyper-client",
+        feature = "reqwest-client",
+        feature = "reqwest-blocking-client"
+    ))]
```
Comment (TommyCpp) [inline opentelemetry-otlp/tests/integration_test/tests/logs.rs:23]: Would be nice to combine those two though. Maybe we can use [cfg_if](https://docs.rs/cfg-if/latest/cfg_if) it's dev only so it's easier to introduce the dependencies

**Review by cijothomas — APPROVED** [review]
Love seeing the integration tests! Thanks!

**Pull request merged**